From ef5ae688788dacb725a64bf6d723f245d3c81529 Mon Sep 17 00:00:00 2001
From: Wes Campaigne <wes@tulip.io>
Date: Tue, 12 Dec 2023 08:02:37 -0500
Subject: [PATCH 1/3] Link to a monolithic libnode instead of libv8*

---
 config.m4 | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/config.m4 b/config.m4
index 9cb3b16c..ffc3b64e 100644
--- a/config.m4
+++ b/config.m4
@@ -3,7 +3,7 @@ PHP_ARG_WITH(v8js, for V8 Javascript Engine,
 
 if test "$PHP_V8JS" != "no"; then
   SEARCH_PATH="/usr/local /usr"
-  SEARCH_FOR="$PHP_LIBDIR/libv8.$SHLIB_SUFFIX_NAME"
+  SEARCH_FOR="$PHP_LIBDIR/libnode.$SHLIB_SUFFIX_NAME"
   
   if test -r $PHP_V8JS/$SEARCH_FOR; then
     case $host_os in
@@ -33,7 +33,7 @@ if test "$PHP_V8JS" != "no"; then
   fi
 
   PHP_ADD_INCLUDE($V8_DIR/include)
-  PHP_ADD_LIBRARY_WITH_PATH(v8, $V8_DIR/$PHP_LIBDIR, V8JS_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(node, $V8_DIR/$PHP_LIBDIR, V8JS_SHARED_LIBADD)
   PHP_SUBST(V8JS_SHARED_LIBADD)
   PHP_REQUIRE_CXX()
 
@@ -95,20 +95,20 @@ if test "$PHP_V8JS" != "no"; then
   AC_MSG_CHECKING([for libv8_libplatform])
   AC_DEFUN([V8_CHECK_LINK], [
     save_LIBS="$LIBS"
-    LIBS="$LIBS $1 -lv8_libplatform -lv8"
+    LIBS="$LIBS $1 -lnode"
     AC_LINK_IFELSE([AC_LANG_PROGRAM([
       #include <libplatform/libplatform.h>
     ], [ v8::platform::NewDefaultPlatform(); ])], [
       dnl libv8_libplatform.so found
       AC_MSG_RESULT(found)
-      V8JS_SHARED_LIBADD="$1 -lv8_libplatform $V8JS_SHARED_LIBADD"
+      V8JS_SHARED_LIBADD="$1 -lnode $V8JS_SHARED_LIBADD"
         $3
     ], [ $4 ])
     LIBS="$save_LIBS"
   ])
 
   V8_CHECK_LINK([], [], [], [
-    V8_CHECK_LINK([-lv8_libbase], [], [], [
+    V8_CHECK_LINK([-lnode], [], [], [
       AC_MSG_ERROR([could not find libv8_libplatform library])
     ])
   ])

From e747e3f67dbb5f45006bc49c3f37bebbbcc68b5d Mon Sep 17 00:00:00 2001
From: Wes Campaigne <wes@tulip.io>
Date: Tue, 12 Dec 2023 08:03:09 -0500
Subject: [PATCH 2/3] Add missing #include <stdexcept> ?

---
 v8js_object_export.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/v8js_object_export.cc b/v8js_object_export.cc
index 87a88b66..61ba0c95 100644
--- a/v8js_object_export.cc
+++ b/v8js_object_export.cc
@@ -16,6 +16,8 @@
 #include "config.h"
 #endif
 
+#include <stdexcept>
+
 #include "php_v8js_macros.h"
 #include "v8js_array_access.h"
 #include "v8js_exceptions.h"

From 16c5bad763c9d54bb3517bb7dd549f906c349c4c Mon Sep 17 00:00:00 2001
From: Wes Campaigne <wes@tulip.io>
Date: Wed, 17 Jan 2024 17:42:00 -0500
Subject: [PATCH 3/3] Make the "timelimit" operation more expensive so it
 doesn't cause a spurious failure if the computer is too fast

---
 tests/time_limit.phpt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/time_limit.phpt b/tests/time_limit.phpt
index 4cef93f8..f6757e5e 100644
--- a/tests/time_limit.phpt
+++ b/tests/time_limit.phpt
@@ -13,7 +13,7 @@ if (getenv("SKIP_SLOW_TESTS")) {
 
 $JS = <<< EOT
 var text = "abcdefghijklmnopqrstuvwyxz0123456789";
-for (var i = 0; i < 10000000; ++i) {
+for (var i = 0; i < 1000000000; ++i) {
     var encoded = encodeURI(text);
 }
 EOT;
